python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2113,6 +2117,13 @@ def get_tick_retention_settings(
         default_tick_settings = get_default_tick_retention_settings(instigator_type)
         return get_tick_retention_settings(tick_settings, default_tick_settings)
 
+    def inject_secrets_into_environment(self, location_name: Optional[str]):
+        # Inject secrets into environment, precedence to first in the list
+        for secret_loader in reversed(self._secrets_loaders):
+            new_env = secret_loader.load_secrets(location_name)
+            for k, v in new_env.items():
+                os.environ[k] = v
```

Comment:
would be good to output debug info about observed changes after invoking each hook - "these environment variables were added / changed (removed?)"